content/learn/book/migration-guides/0.8-0.9/_index.md

```diff
@@ -0,0 +1,825 @@
+// Old (0.8):
+commands
+  .spawn()
+  .insert_bundle((A, B, C));
+// New (0.9)
+commands.spawn((A, B, C));
+
+// Old (0.8):
+commands.spawn_bundle((A, B, C));
+// New (0.9)
+commands.spawn((A, B, C));
+
+// Old (0.8):
+let entity = commands.spawn().id();
+// New (0.9)
+let entity = commands.spawn_empty().id();
+
+// Old (0.8)
+let entity = world.spawn().id();
+// New (0.9)
+let entity = world.spawn_empty();
+```
+
+### [Accept Bundles for insert and remove. Deprecate `insert`/`remove_bundle`](https://github.com/bevyengine/bevy/pull/6039)
+
+Replace `insert_bundle` with `insert`:
+
+```rust
+// Old (0.8)
+commands.spawn().insert_bundle(SomeBundle::default());
+// New (0.9)
+commands.spawn_empty().insert(SomeBundle::default());
+```
+
+Replace `remove_bundle` with `remove`:
+
+```rust
+// Old (0.8)
+commands.entity(some_entity).remove_bundle::<SomeBundle>();
+// New (0.9)
+commands.entity(some_entity).remove::<SomeBundle>();
+```
+
+Replace `remove_bundle_intersection` with `remove_intersection`:
+
+```rust
+// Old (0.8)
+world.entity_mut(some_entity).remove_bundle_intersection::<SomeBundle>();
+// New (0.9)
+world.entity_mut(some_entity).remove_intersection::<SomeBundle>();
+```
+
+Consider consolidating as many operations as possible to improve ergonomics and cut down on archetype moves:
+
+```rust
+// Old (0.8)
+commands.spawn()
+  .insert_bundle(SomeBundle::default())
+  .insert(SomeComponent);
+
+// New (0.9) - Option 1
+commands.spawn_empty().insert((
+  SomeBundle::default(),
+  SomeComponent,
+))
+
+// New (0.9) - Option 2
+commands.spawn((
+  SomeBundle::default(),
+  SomeComponent,
+))
+```
+
+### [Implement `Bundle` for `Component`. Use `Bundle` tuples for insertion](https://github.com/bevyengine/bevy/pull/2975)
+
+In `derive(Bundle)`, the `bundle` attribute has been removed. Nested bundles are now collapsed automatically. You should remove `#[bundle]` attributes.
+
+### [Replace the `bool` argument of `Timer` with `TimerMode`](https://github.com/bevyengine/bevy/pull/6247)
+
+* Replace `Timer::new(duration, false)` with `Timer::new(duration, TimerMode::Once)`.
+* Replace `Timer::new(duration, true)` with `Timer::new(duration, TimerMode::Repeating)`.
+* Replace `Timer::from_seconds(seconds, false)` with `Timer::from_seconds(seconds, TimerMode::Once)`.
+* Replace `Timer::from_seconds(seconds, true)` with `Timer::from_seconds(seconds, TimerMode::Repeating)`.
+* Change `timer.repeating()` to `timer.mode() == TimerMode::Repeating`.
+
+### [Add global time scaling](https://github.com/bevyengine/bevy/pull/5752)
+
+* `time.time_since_startup()` -> `time.elapsed()`
+* `time.seconds_since_startup()` -> `time.elapsed_seconds_f64()`
+* `time.seconds_since_startup_wrapped_f32()` -> `time.elapsed_seconds_wrapped()`
+
+If you aren’t sure which to use, most systems should continue to use “scaled” time (e.g. `time.delta_seconds()`). The realtime “unscaled” time measurements (e.g. `time.raw_delta_seconds()`) are mostly for debugging and profiling.

```

Comment:
~~~suggestion
Some `Time` methods were renamed for consistency after new methods were added.

```rust
// Old (Bevy 0.8)
let dur: Duration = time.time_since_startup();
let secs: f32 = time.time_since_startup().as_secs_f32();
let secs: f64 = time.seconds_since_startup();

// New (Bevy 0.9)
let dur: Duration = time.elapsed();
let secs: f32 = time.elapsed_seconds();
let secs: f64 = time.elapsed_seconds_f64();
```
~~~

Rewrote as a code block and made sure both chunks of code compile.

Added a line with `elapsed_seconds` because the previous suggestion could have lead users to do `elapsed_seconds_f64() as f32` unnecessarily.

The paragraph that was there previously seemed irrelevant to migration so I removed it.

`seconds_since_startup_wrapped_f32` didn't exist in bevy 0.8, so I don't think we need to mention it.

Reply:
Sorry for the previous notification. I botched the suggestion and didn't have time to fix it.

Reply:
I think we should keep the recommendation to use scaled time here to give more context.

Reply:
After re-reading several times, I agree. I updated the suggestion.

I reworked it a little bit because "If you aren’t sure which to use" seemed completely out of context.